app/controllers/application_controller.rb

```diff
@@ -10,4 +13,15 @@ class ApplicationController < ActionController::API
       render json: { error: error }, status: status
     end
   end
+
+  private
+
+  def authenticate
+    authenticate_with_http_token do |token, options|
+      user = User.find_by(id: options[:user_id])
+      @current_user = user&.authenticate_auth_token(token)
```

Comment:
Small question about the variable scope:
does `@current_user` needs to be an instance var in our service? 
I know it tends to be the default but I suspect we only need it  in `authenticate` method scope and don't use it anywhere else...

Reply:
Yep, that is very good point about it not being needed outside the scope of this check. Ill refactor it.